event_model/schemas/run_start.json

```diff
@@ -4,6 +4,34 @@
             "title" : "data_type",
             "patternProperties": {"^([^./]+)$": {"$ref": "#/definitions/data_type"}},
             "additionalProperties": false
+        },
+        "data_mapping": {
+            "description": "Where to get the data from",
+            "type": "object",
+            "properties" : {
+                "stream": {"type": "string"},
+                "location": {"enum" : ["event", "configuration"]},
```

Comment:
What if it's in the stop document? Or elsewhere in the EventDescriptor, such as the `source` or `shape`? Would a dotted object representation be simpler and more comprehensive?

Reply:
Being in the start document is a far more compelling argument.

I'm skeptical of embracing the dot-ness (as we have previously agreed ed that dot access on dicts is not great) so the code to munge that back to something we can actually use will be annoying, but on the other hand we can write the function once and stuff it it databroker).